### DISCUSSION TITLE
I fixed some of the Bugs of this Plugin, find my fork here

### DISCUSSION THREAD

**Discussion by Fransferdy** [AndreaCatania/RuntimeSkeletalMeshGenerator#23] [General]
Hello everyone, if you look around you'll see I have worked a bit on this lib, this repository is somewhat abandoned, I have made a fork with the fixes I and some others developers have made to this plugin, you may find it here:
https://github.com/Fransferdy/RuntimeSkeletalMeshGenerator

Works on UE 5.1 (not on UE 5.2,5.3)

**Comment by matiasgql** [comment]
It works on UE5.1 but the mesh doesn't render in 5.2 or 5.3 (I tried the last 3 weeks and I can't find the bug)

**Reply by Fransferdy** [reply]
I spent the whole day trying to get this to work on UE 5.3. And had the same result as you.
The console logs that there is an issue with LOD 0 Material. Frankly I would just stick with UE 5.1 in case you want the feature of this plugin. 5.1 is already a fairly powerful game engine. Also if you start a project in an Engine version, you better finish the project in that same version, if you don't want to have a lot of headaches.

**Reply by spood** [reply]
> The console logs that there is an issue with LOD 0 Material. 

I tracked that down to this change https://github.com/Fransferdy/RuntimeSkeletalMeshGenerator/commit/1dac96e74ac0ae336a05e0f9ef1d597c7bed3416 
Reverting it resolved the LOD 0 Material error I was seeing. Not sure if it was an incomplete change or what but the Surface MaterialIndex had massive random numbers in it that I presume should have been getting initialized from somewhere, but that wasn't included in the change.

I made some changes to get it running on UE5.4 as well but I see this same issue where the mesh doesn't render. I can see it generated a skeletal mesh asset, and previewing it shows the right number of trianges and the material sections look correct, but there's no wireframe or any errors. Tried digging into the changes to the engine between 5.1 and 5.2 but I couldn't see anything that could explain why this no longer works. I don't know how to debug this any further 😢 but I would like to get it working.

**Reply by Fransferdy** [reply]
Frankly, I didn't even test that change(I google translated the commit comments, saw it was a bug fix and called it a day), sorry, I will revert it in my main branch as well. I would also like to get this to work, I couldn't manage to fix it myself either and I don't have the time or motivation to do so anymore.
But one idea that I had it might be easier and more stable, to just have one base skeletal model, which is basically invisible(maybe 3 verts, one face) with 2 bones, and then adapt this plugin, to just copy the surfaces/materials/bones/etc/, on top of this simple model, so that whenever the engine changes, the plugin won't break, because the base data will be populated by that "InEngine" base model, so you would be always merging that base model with you "current" model.
edit: for all that it is worth, I tried the merging ideia, but it crashed during the merge as well.

**Reply by rtjia** [reply]
> > The console logs that there is an issue with LOD 0 Material.
> 
> I tracked that down to this change [Fransferdy@1dac96e](https://github.com/Fransferdy/RuntimeSkeletalMeshGenerator/commit/1dac96e74ac0ae336a05e0f9ef1d597c7bed3416) Reverting it resolved the LOD 0 Material error I was seeing. Not sure if it was an incomplete change or what but the Surface MaterialIndex had massive random numbers in it that I presume should have been getting initialized from somewhere, but that wasn't included in the change.
> 
> I made some changes to get it running on UE5.4 as well but I see this same issue where the mesh doesn't render. I can see it generated a skeletal mesh asset, and previewing it shows the right number of trianges and the material sections look correct, but there's no wireframe or any errors. Tried digging into the changes to the engine between 5.1 and 5.2 but I couldn't see anything that could explain why this no longer works. I don't know how to debug this any further 😢 but I would like to get it working.

In UE5.4 , my material have some error, how to config material in this plugin?
Thanks alot.
